File: typescript/docs_code_snippets/quickstart-audio.ts
import { readFileSync, writeFileSync } from "node:fs";
import { z } from "zod";
import { Opper } from "opperai";

const opper = new Opper();

const audioB64 = readFileSync("input.wav").toString("base64");

const result = await opper.call("audio-summarize", {
  instructions: "Listen to the audio and provide a text summary and an audio narration of the summary",
  input: { audio: audioB64 },
  input_schema: z.object({
    audio: z.string().describe("base64-encoded WAV audio"),
  }),
  output_schema: z.object({
    summary: z.string(),
    audio_summary: z.string().describe("base64-encoded WAV audio"),
  }),
});

console.log("Summary:", result.data.summary);
writeFileSync("summary.wav", Buffer.from(result.data.audio_summary, "base64"));

File: python/docs_code_snippets/quickstart-audio.py
from opperai import Opper

opper = Opper()

# Setup: generate audio via TTS
tts = opper.text_to_speech("docs-tts", text="Hello, this is a test.")
audio_b64 = tts.data["audio"]

# --- docs ---
result = opper.call(
    "audio-summarize",
    instructions="Listen to the audio and provide a text summary and an audio narration of the summary",
    input={"audio": audio_b64},
    input_schema={
        "type": "object",
        "properties": {
            "audio": {"type": "string", "contentMediaType": "audio/wav", "contentEncoding": "base64"}
        },
    },
    output_schema={
        "type": "object",
        "properties": {
            "summary": {"type": "string"},
            "audio_summary": {"type": "string", "contentMediaType": "audio/wav", "contentEncoding": "base64"},
        },
    },
)

print("Summary:", result.data["summary"])
# --- /docs ---
